### PR TITLE
feat(cli): support teamsfx provision manifest

### PR DESCRIPTION
**Pull request by 1yefuwang1**
Related task: https://msazure.visualstudio.com/Microsoft%20Teams%20Extensibility/_sprints/taskboard/Teams%20Extensibility%20E2E%20Team/Microsoft%20Teams%20Extensibility/CY22-2.1?workitem=13294461

Implement `teamsfx provision manifest` command

E2E TEST: packages/cli/tests/e2e/manifest/provision.test.ts

### CODE REVIEW

File: packages/cli/src/cmds/provision.ts [hunk 1]
```diff
@@ -20,18 +20,72 @@ import {
 import HelpParamGenerator from "../helpParamGenerator";
 import { sqlPasswordConfirmQuestionName, sqlPasswordQustionName } from "../constants";
 
+export class ProvisionManifest extends YargsCommand {
+  public readonly commandHead = "manifest";
+  public readonly command = this.commandHead;
+  public readonly description =
+    "Provision a Teams App in Teams Developer portoal with corresponding information specified in the given manifest file";
```
Comment (nliu-ms) [inline packages/cli/src/cmds/provision.ts:27]: portoal -> Portal

**Review by nliu-ms — APPROVED** [review]

File: packages/cli/src/cmds/provision.ts (hunk 1 above)
Comment (1yefuwang1) [reply packages/cli/src/cmds/provision.ts]: Nice catch! Fixed, thx.

**Review by jayzhang — DISMISSED** [review]

**Review by jayzhang — APPROVED** [review]

**Review by HuihuiWu-Microsoft — APPROVED** [review]

File: packages/fx-core/src/plugins/resource/appstudio/v3/plugin.ts [hunk 2]
```diff
@@ -84,11 +84,11 @@ export class AppStudioPluginImpl {
         appStudioToken!,
         ctx.logProvider
       );
-      ctx.logProvider?.info(`Teams app created: ${appDefinition.appId}`);
+      ctx.userInteraction?.showMessage("info", `Teams app created: ${appDefinition.appId}`, false);
```
Comment (HuihuiWu-Microsoft) [inline packages/fx-core/src/plugins/resource/appstudio/v3/plugin.ts:87]: For VSC, userInteraction will show the notification dialog which means the dialog(app created/updated) will be displayed when user provision, publish, localdebug, .... Just want to make sure whether that's expected behavior.

Comment (1yefuwang1) [reply packages/fx-core/src/plugins/resource/appstudio/v3/plugin.ts:87]: Just checked with @nliu-ms that it's ok.

**Review by Alive-Fish — APPROVED** [review]

**Pull request merged**